File: articles/decision-under-uncertainty-and-computational-risk/calculators/r/expected_value_calculator.R
args <- commandArgs(trailingOnly = TRUE)
values <- as.numeric(args)
if (length(values) != 4) stop("Usage: Rscript expected_value_calculator.R probability benefit loss cost")
p <- max(0, min(1, values[1]))
expected_value <- p * values[2] - p * values[3] - values[4]
cat(sprintf("probability=%.6f\n", p))
cat(sprintf("expected_value=%.6f\n", expected_value))

File: articles/decision-under-uncertainty-and-computational-risk/calculators/python/expected_value_calculator.py
from __future__ import annotations
import argparse


def main() -> None:
    parser = argparse.ArgumentParser(description="Compute expected net value under uncertainty.")
    parser.add_argument("--probability", type=float, required=True)
    parser.add_argument("--benefit", type=float, required=True)
    parser.add_argument("--loss", type=float, required=True)
    parser.add_argument("--cost", type=float, required=True)
    args = parser.parse_args()
    p = max(0.0, min(1.0, args.probability))
    expected_value = p * args.benefit - p * args.loss - args.cost
    print(f"probability={p:.6f}")
    print(f"expected_value={expected_value:.6f}")


if __name__ == "__main__":
    main()
